### ISSUE TITLE
Rib generation issues:

### ISSUE THREAD

**Issue by bsavery** [bsavery/PRMan-for-Blender#125]
This thread is for posting rib generation issues.  I have a minor overhaul of rib generation in mind.  Probably about a days worth of work.  Basically there are two main points:

1.  We already export data blocks to their own rib archives.  We could combine the caching of meshes for motion blur purposes along with exporting the data blocks to minimize walking through object list.  For speed and efficiency.

2.  The "Cache rib generation" feature works ok for objects, could be better for particle system and smoke, etc.  In fact I'm not sure it works correctly at all for hair or smoke.  

Couple other notes:  Currently I'm not sure if two objects, same data block but one has a modifier and one does not works correctly.  We just output one data block.  We also should support the particle system render options.  As well as check for layers when outputting particle systems.  

Basically the main goals are:  Rewrite the system that marks objects as "dirty" for exporting rib.  Export data block archives when we get the motion and export the "read archives" of the data block in export_objects.
